### PR TITLE
#3242 if structures rewritten in order to avoid IDA failure, useless …

### PR DESCRIPTION
**Pull request by ErwanGuichard49**
…parameters removed from .par

Veuillez saisir le message de validation pour vos modifications. Les lignes

**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/Ac7c.mo [hunk 1]
```diff
@@ -206,73 +206,57 @@ equation
...
-    min2.u[2] = min2.u[1];
+    min1.u[2] = if PositionScl == 2 then min(UOelPu, USclOelPu) else min1.u[1];
+    min2.u[2] = if PositionScl == 3 then USclOelPu else min2.u[1];
     min3.u1 = min3.u2;
   elseif PositionOel == 3 then
     sum1.u[3] = 0;
-    min1.u[2] = min1.u[1];
-    min2.u[2] = UOelPu;
+    min1.u[2] = if PositionScl == 2 then USclOelPu else min1.u[1];
+    min2.u[2] = if PositionScl == 3 then min(UOelPu, USclOelPu) else min2.u[1];
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/Ac7c.mo:220]: else UOelPu

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/St6c.mo [hunk 2]
```diff
@@ -160,93 +160,73 @@ model St6c "IEEE exciter type ST6C model"
...
     sum1.u[2] = UOelPu;
-    min1.u[2] = min1.u[1];
+    min1.u[2] = if PositionScl == 2 then USclOelPu else min1.u[1];
     sum2.u[3] = 0;
-    min2.u[2] = min2.u[1];
+    min2.u[2] = if PositionScl == 4 then USclOelPu else min2.u[1];
   elseif PositionOel == 2 then
     sum1.u[2] = 0;
-    min1.u[2] = UOelPu;
+    min1.u[2] = if PositionScl == 2 then min(UOelPu, USclOelPu) else min1.u[1];
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/St6c.mo:168]: else UOelPu

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/St6c.mo [hunk 3]
```diff
@@ -160,93 +160,73 @@ model St6c "IEEE exciter type ST6C model"
...
     sum2.u[3] = UOelPu;
-    min2.u[2] = min2.u[1];
+    min2.u[2] = if PositionScl == 4 then USclOelPu else min2.u[1];
   elseif PositionOel == 4 then
     sum1.u[2] = 0;
-    min1.u[2] = min1.u[1];
+    min1.u[2] = if PositionScl == 2 then USclOelPu else min1.u[1];
     sum2.u[3] = 0;
-    min2.u[2] = UOelPu;
+    min2.u[2] = if PositionScl == 4 then min(UOelPu, USclOelPu) else min2.u[1];
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/St6c.mo:180]: else UOelPu

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/St6c.mo [hunk 4]
```diff
@@ -160,93 +160,73 @@ model St6c "IEEE exciter type ST6C model"
...
     sum1.u[3] = UUelPu;
-    max1.u[2] = max1.u[1];
+    max1.u[2] = if PositionScl == 2 then USclUelPu else max1.u[1];
     sum2.u[4] = 0;
-    max2.u[2] = max2.u[1];
+    max2.u[2] = if PositionScl == 4 then USclUelPu else max2.u[1];
   elseif PositionUel == 2 then
     sum1.u[3] = 0;
-    max1.u[2] = UUelPu;
+    max1.u[2] = if PositionScl == 2 then max(UUelPu, USclUelPu) else max1.u[1];
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/St6c.mo:195]: else UUelPu

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/St6c.mo [hunk 5]
```diff
@@ -160,93 +160,73 @@ model St6c "IEEE exciter type ST6C model"
...
     sum2.u[4] = UUelPu;
-    max2.u[2] = max2.u[1];
+    max2.u[2] = if PositionScl == 4 then USclUelPu else max2.u[1];
   elseif PositionUel == 4 then
     sum1.u[3] = 0;
-    max1.u[2] = max1.u[1];
+    max1.u[2] = if PositionScl == 2 then USclUelPu else max1.u[1];
     sum2.u[4] = 0;
-    max2.u[2] = UUelPu;
+    max2.u[2] = if PositionScl == 4 then max(UUelPu, USclUelPu) else max2.u[1];
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/St6c.mo:207]: else UUelPu

**Review by joyelfeghali — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/Ac7c.mo [hunk 6]
```diff
@@ -206,73 +206,57 @@ equation
...
-    min1.u[2] = min1.u[1];
-    min2.u[2] = min2.u[1];
+    min1.u[2] = if PositionScl == 2 then USclOelPu else min1.u[1];
+    min2.u[2] = if PositionScl == 3 then USclOelPu else min2.u[1];
     min3.u1 = min3.u2;
   elseif PositionOel == 2 then
     sum1.u[3] = 0;
-    min1.u[2] = UOelPu;
-    min2.u[2] = min2.u[1];
+    min1.u[2] = if PositionScl == 2 then min(UOelPu, USclOelPu) else min1.u[1];
```
[the remark this answers is not in the document]
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/Ac7c.mo:214]: Indeed

**Review by joyelfeghali — DISMISSED** [review]

**Comment by ErwanGuichard49** [comment]
I am checking the nrt, please do not merge before they have passed.

**Comment by ErwanGuichard49** [comment]
The nrt references have been updated. The branch can thus be merged.

**Review by joyelfeghali — APPROVED** [review]

**Pull request merged**